reframe/core/modules.py

```diff
@@ -391,7 +394,14 @@ def __init__(self):
         if version_match is None or tcl_version_match is None:
             raise ConfigError('could not find a sane Tmod installation')
 
-        self._version = version_match.group(1)
+        version = version_match.group(1)
+        ver_major, ver_minor, *_ = [int(v) for v in version.split('.')]
```

Comment:
You should also guard here against `ValueError` thrown by the conversion to `int`.
In such case, you should raise an error that you couldn't parse the version information.